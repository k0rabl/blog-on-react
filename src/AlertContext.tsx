import React, { Component } from 'react'

interface IState {
  isOpen: boolean
  type: string
}

const AlertContext = React.createContext({
    isOpen: false,
    type: '',
    setOpen: (c: boolean) => {},
    setType: (c: string) => {}
  })


class AlertProvider extends Component {
  // Context state
  state: IState = {
    isOpen: false,
    type: ''
  }
 

  // Method to update state
  setOpen = (isOpen: boolean) => {
    this.setState((prevState) => ({ isOpen }))
  }

  setType = (type: string) => {
    this.setState((prevState) => ({ type }))
  }

  render() {
    const { children } = this.props
    const { isOpen, type } = this.state
    const { setType, setOpen } = this

    return (
      <AlertContext.Provider
        value={{
          isOpen,
          type,
          setOpen,
          setType,
        }}
      >
        {children}
      </AlertContext.Provider>
    )
  }
}

export default AlertContext

export { AlertProvider }