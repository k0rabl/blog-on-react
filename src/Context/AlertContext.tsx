import React, { Component } from 'react' 

interface IState {
  isOpen: boolean
  message: string
}

const AlertContext = React.createContext({
    isOpen: false,
    message: '',
    setOpen: (c: boolean) => {},
    setType: (c: string) => {}
  })


class AlertProvider extends Component {
  // Context state
  state: IState = {
    isOpen: false,
    message: ''
  }
 

  // Method to update state
  setOpen = (isOpen: boolean) => {
    this.setState((prevState) => ({ isOpen }))
  }

  setType = (message: string) => {
    this.setState((prevState) => ({ message }))
  }

  render() {
    const { children } = this.props
    const { isOpen, message } = this.state
    const { setType, setOpen } = this

    return (
      <AlertContext.Provider
        value={{
          isOpen,
          message,
          setOpen,
          setType,
        }}
      >
        {children}
      </AlertContext.Provider>
    )
  }
}

export default AlertContext

export { AlertProvider }