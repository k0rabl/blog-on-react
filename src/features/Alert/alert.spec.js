import React from "react"
import { Alert } from './index'
import { mount } from 'enzyme';

import {
  BrowserRouter as Router,
} from "react-router-dom"


import { AlertProvider } from "../../Context/AlertContext";

describe('<Alert />', () => {
  let component
  beforeEach(() => {
    
    jest.spyOn(React, 'useEffect').mockImplementation(f => f())
    component = mount(
      <AlertProvider value={{isOpen: true, message: ''}}>
        <Router>
          <Alert />
        </Router>
      </AlertProvider>
    )
  })

  it('+++ render the component', () => {
    expect(component.find(Alert).length).toEqual(1)
    expect(component).toMatchSnapshot()
  })

  
  it('+++ check alert is open', () => {
    setTimeout(() => {
      expect(component.prop('isOpen')).toBeFalsy()
    }, 5100);
  })
})