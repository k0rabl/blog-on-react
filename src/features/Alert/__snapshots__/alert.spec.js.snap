// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> +++ render the component 1`] = `
<AlertProvider
  value={
    Object {
      "isOpen": true,
      "message": "",
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Alert>
        <div
          className="alert "
        >
          <div
            className="content"
          />
          <div
            className="close"
          />
        </div>
      </Alert>
    </Router>
  </BrowserRouter>
</AlertProvider>
`;
